nvflare/app_common/executors/task_script_runner.py

```diff
@@ -61,14 +61,29 @@ def get_sys_argv(self):
         return [self.script_path] + args_list
 
     def get_script_full_path(self, script_path) -> str:
-        target_files = None
+        target_file = None
+        script_filename = os.path.basename(script_path)
+        script_dirs = os.path.dirname(script_path)
+
         for r, dirs, files in os.walk(os.getcwd()):
-            target_files = [os.path.join(r, f) for f in files if f == script_path]
-            if target_files:
+            for f in files:
+                absolute_path = os.path.join(r, f)
+                if absolute_path.endswith(script_path):
+                    parent_dir = absolute_path[: absolute_path.find(script_path)].rstrip(os.sep)
+                    if os.path.isdir(parent_dir):
+                        target_file = absolute_path
+                        break
+
+                if not script_dirs and f == script_filename:
+                    target_file = absolute_path
+                    break
```

Comment:
I am not quite sure what's the purpose of doing these additional checks. But this unit test will fail. Do you want to support the absolute path script search?

    def test_app_scripts_and_args(self):
        curr_dir = os.getcwd()
        script_path = os.path.join(curr_dir, "nvflare/cli.py")
        script_args = "--batch_size 4"
        wrapper = TaskScriptRunner(script_path=script_path, script_args=script_args)

        self.assertTrue(wrapper.script_path.endswith(script_path))
        self.assertEqual(wrapper.get_sys_argv(), [os.path.join(curr_dir, "nvflare", "cli.py"), "--batch_size", "4"])

Reply:
Actually, we don't support the absolute path script in the add_ext_script(). so no need to handle this case.